Finally Block: 
​
finally block is followed by try or catch blocks.
​
try{
​
}
catch(Exception e){
​
}
finally{
​
}
​
​
try{
​
}
finally{
​
}
​
​

import java.io.FileNotFoundException;
import java.io.FileReader;

public class FinallyDemo {
	 public static void main(String[] args)  {
		 FileReader fr= null;
		  try {
			 fr= new FileReader("D:\\1coding\\3.java\\java code\\7nov\\ExceptionHandling\\src\\CheckedExceptionDemo.java");
			 System.out.println("file read");
			
			 
		} catch (Exception e) {
			 
			System.out.println("catch:" + e);
			// Log info into Database
			
			
		}
		  finally {
			  System.out.println("inside finally");
			  try {
			  fr.close();
			  System.out.println("fr closed");
			  }
			  catch(Exception e) {
				  System.out.println("finally exception:"+e);
			  }
			  
		  }
		  System.out.println("Statement 1");
		  System.out.println("Statement 2");
		 
	}
}